### DISCUSSION TITLE
Error while converting pdf to png using ImageMagick

### DISCUSSION THREAD

**Discussion by pulkitbhati** [ImageMagick/ImageMagick#2689] [Help]
Hi,

There are pdf files that I am converting to png images using ImageMagick. It is working fine for maximum documents.But it is getting stuck for few documents that have various different type of font formats.
I am getting below error while converting those pdf documents to png images :

 **** Error: Invalid (0 scaling) text matrix for Tm ****
               Output may be incorrect.
   **** Error: Invalid (0 scaling) text matrix for Tm ****
               Output may be incorrect.
   **** Error: Invalid (0 scaling) text matrix for Tm ****
               Output may be incorrect.
   **** Error: Invalid (0 scaling) text matrix for Tm ****
               Output may be incorrect.

Due to this my java process is getting stuck and not able to proceed.

Please suggest how this error can be correct and how such errors can be caught and handled so that my java process does not get stuck due to this.

**Comment by snibgo** [comment]
What versions of IM and Ghostscript, on what platform?

IM delegates rasterization of PDF to Ghostscript. I suppose GS is reporting those errors, which look like the PDF is scaling text to zero size.

What is your IM command? If you convert each file in a separate command, your java script can check for IM failing. When it fails, the script could report the problem then move on to the next PDF. I don't use java, and can't advise on that.

**Comment by pulkitbhati** [comment]
GPL Ghostscript 9.53.3 (2020-10-01)
Version: ImageMagick 7.0.10-33 Q16

Command :

/usr/local/bin/convert -density 150 -bordercolor white -border 1x1 -quality 100 9_59_11_3000000_0_15_6_59_935000000_434_9g37e285e6955b6.pdf /home/users/pulkitbhatia/testimage/4135797.png

**Comment by snibgo** [comment]
For IM v7, I suggest using "magick" instead of "convert".

Okay, so your java script should check the return code from the "magick" or "convert" command. I don't know how to do that in java.
